src/collector.ts

```diff
@@ -105,9 +105,38 @@ class NaiveGomodParser {
     dependencies: Array<IDependency>;
 
     static parseDependencies(contents:string, goImports: Set<string>): Array<IDependency> {
+        let replaceDependencies = new Map();
+
+        function replaceMap(line: string, index: number) {
+            // skip any text after '//'
+            if (line.includes("//")) {
+                line = line.split("//")[0];
+            }
+            // split the replace statements by '=>'
+            const parts: Array<string> = line.replace('replace', '').replace('(', '').replace(')', '').trim().split('=>');
+            const replaceWith: Array<string> = (parts[1] || '').trim().split(' ');
+            regExp.lastIndex = 0;
+            const replaceWithVersion = regExp.exec(parts[1]);
+
+            // Skip lines without final version string
+            if (replaceWithVersion && replaceWithVersion.length > 0) {
+                const replaceTo: Array<string> = (parts[0] || '').trim().split(' ');
+                regExp.lastIndex = 0;
+                const replaceToVersion = regExp.exec(replaceTo[1]);
+                const replaceWithIndex = line.lastIndexOf(parts[1]);
+                const replaceEntry: IKeyValueEntry = new KeyValueEntry(replaceWith[0].trim(), { line: 0, column: 0 });
+                replaceEntry.value = new Variant(ValueType.String, 'v' + replaceWithVersion[0]);
+                replaceEntry.value_position = { line: index + 1, column: (replaceWithIndex + replaceWithVersion.index) };
+                const replaceDependency = new Dependency(replaceEntry);
+                const isReplaceToVersion: boolean = replaceToVersion && replaceToVersion.length > 0;
+                replaceDependencies.set(replaceTo[0].trim() + (isReplaceToVersion ? ('@v' + replaceToVersion[0]) : ''), replaceDependency);
+            }
+        }
+
         let goModDeps = contents.split("\n").reduce((dependencies, line, index) => {
-            // Ignore "replace" lines
-            if (!line.includes("=>")) {
+            if (line.includes("=>")) {
+                replaceMap(line, index);
```

Comment:
Better name for this function can be checkAndBuildReplaceMap()

Reply:
I prefer applyReplaceDirective(..)

Reply:
@arajkumar Here replace map is build by this function during go.mod traversal. Apply of map happens later after this loop.

Reply:
what about generateReplaceMap()?